### DISCUSSION TITLE
Untranslated texts in anguage file (Language/es/Auth)

### DISCUSSION THREAD

**Discussion by wolverine4277** [codeigniter4/shield#1209] [Q&A]
Hi, I recently started with Shield and found some untranslated texts:

```
'invalidEmail'          => '(To be translated) Unable to verify the email address "{0}" matches the email on record.',
'invalidJWT'     => '(To be translated) The token is invalid.',
'expiredJWT'     => '(To be translated) The token has expired.',
'beforeValidJWT' => '(To be translated) The token is not yet available.',
'token'           => '(To be translated) Token',
'magicLinkDisabled'  => '(To be translated) Use of MagicLink is currently not allowed.',
```

The texts translated into Spanish would be:

```
'invalidEmail'          => 'No se ha podido verificar si la dirección de correo electrónico "{0}" coincide con la registrada.',
'invalidJWT'     => 'El token es inválido.',
'expiredJWT'     => 'El token ha expirado.',
'beforeValidJWT' => 'El token todavía no se encuenta disponible.',
'token'           => 'Token',
'magicLinkDisabled'  => 'El uso de MagicLink no está permitido actualmente.',
```

If the message should go somewhere else, feel free to move it.
Greetings.

**Comment by datamweb** [comment]
Hi,
To make changes to a public project, you need to submit a **Pull Request (PR)**. Here’s a complete and simple guide for the process:

### 1. Fork the Project
First, fork the repository where you want to make changes:
1. Go to the [main page of the project’s](https://github.com/codeigniter4/shield) repository on GitHub.
2. Click on the **[Fork](https://github.com/codeigniter4/shield/fork)** button in the top right corner to create a copy of the project in your account.

### 2. Clone the Forked Repository
Once you’ve forked the project, clone the repository to your local machine:
```console
git clone https://github.com/your-username/shield.git
cd shield
```

### 3. Create a New Branch for Your Changes
To work on your changes, create a new branch:
```bash
git switch -c feature-branch-name develop
```

### 4. Make the Changes
Apply the necessary changes to the project files.

### 5. Commit the Changes and Sign with GPG
Before committing your changes, it's important to set up **GPG signing** to ensure your commits are secure and verified.

#### Steps to Set Up GPG Signing:
1. **Generate a GPG key:**
   You’ll first need to generate a GPG key. Run the following in your terminal:
   ```console
   gpg --full-generate-key
   ```
   Follow the on-screen instructions to create your key.

2. **Export Your GPG Public Key:**
   To see your GPG keys, use:
   ```console
   gpg --list-secret-keys --keyid-format LONG
   ```
   Then export the public key:
   ```console
   gpg --armor --export <your-key-id>
   ```
   Add this key to your GitHub account in the **GPG keys** section under your profile settings.

3. **Configure Git to Use Your GPG Key:**
   Set up Git to use your GPG key for signing commits:
   ```console
   git config --global user.signingkey <your-key-id>
   git config --global commit.gpgSign true
   ```

4. **Commit Your Changes:**
   Now, commit your changes and sign them with GPG:
   ```console
   git add .
   git commit -S -m "lang: Description of your changes"
   ```

### 6. Push Your Changes to GitHub
After committing, push your changes to the new branch in your forked repository:
```console
git push origin feature-branch-name
```

### 7. Create a PR (Pull Request)
Now, go to GitHub:
1. Go to your forked repository.
2. GitHub will prompt you to create a PR from your new branch. Click **Compare & pull request**.
3. Add a description of your changes, then click **Create pull request**.

Your PR is now submitted and waiting for review and approval!

Good luck!
